working/macros/feature-specification.md

```diff
@@ -553,6 +553,98 @@ Note, that if the getter were written as `int get y => this.x;`, then a macro
 *would* be allowed to introduce the new getter `x`, because `this.x` could not
 previously be resolved.
 
+## Macro Execution Environment
+
+The execution environment for macros is different from that of normal code.
+Specifically, we have some semantics that we want to uphold (to the extent
+that is feasible) which puts some constraints on the environment.
+
+- No direct access to the host device (except through the `Resource` API).
+- No ability to spawn arbitrary processes or isolates.
+- Can only run synchronous code.
```

Comment:
> (We also have the option of having a separate async builder API that macro authors could choose to implement.)

I would prefer to avoid this option, the single api with `FutureOr<void>` intuitively supports both (that is fwiw what we do in the builder api for `package:build`).

But I agree we have plenty of time to explore this internally, and could stick with a sync api for now.